src/ansi-c/expr2c.cpp

```diff
@@ -1866,6 +1869,27 @@ std::string expr2ct::convert_constant(
       else if(dest=="-NaN")
         dest="-NAN";
     }
+    else if(dest.size() == 4 && (dest[0] == '+' || dest[0] == '-'))
```

Comment:
Combine this and the case above to remove the redundant expression `dest.size() == 4 && (dest[0] == '+' || dest[0] == '-')`

Reply:
I was trying to keep the diff smaller, but will now do as suggested.